### PR TITLE
Improvements to custom styles in rx.markdown

### PR DESCRIPTION
**Pull request by picklelo**
Added option to set custom styles using a component map:

```python
component_map = {
    "h1": lambda text: rx.box(rx.heading(text), rx.divider())
    "p": rx.text
}
```

Then pass it in

```python
source = """
# Hello!

Welcome to my markdown file.
"""

rx.markdown(
    source,
    component_map=component_map
)
```

### CODE REVIEW

**Review by masenf — DISMISSED** [review]
would be cool to add e2e tests for the new var operations in `integration/test_var_operations.py`.

is this enough to fix #1283 too?

File: reflex/components/typography/markdown.py [hunk 1]
```diff
@@ -38,43 +52,18 @@ class Markdown(Component):
 
     is_default = True
 
-    # Custom defined styles for the markdown elements.
-    custom_styles: Dict[str, Style] = {
```
Comment (masenf) [inline reflex/components/typography/markdown.py:42]: we probably should deprecate this first, unless you think no one is using it or it's already broken

Comment (picklelo) [reply reflex/components/typography/markdown.py:42]: Yes agreed, I will add this for the following release I think I can make the old version convert cleanly to the new one.

File: reflex/testing.py [hunk 2]
```diff
@@ -398,7 +398,7 @@ def frontend(self, driver_clz: Optional[Type["WebDriver"]] = None) -> "WebDriver
             )
         if self.frontend_url is None:
             raise RuntimeError("Frontend is not running.")
-        driver = driver_clz() if driver_clz is not None else webdriver.Chrome()
+        driver = driver_clz() if driver_clz is not None else webdriver.Firefox()
```
Comment (masenf) [inline reflex/testing.py:401]: not sure if we want to commit this...

**Review by masenf — APPROVED** [review]

File: reflex/testing.py (hunk 2 above)
Comment (picklelo) [reply reflex/testing.py]: Fixed this, just had it locally for testing

**Review by masenf — APPROVED** [review]

**Pull request merged**